src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/service/cessations/BeneficialOwnerCessationService.java

```diff
@@ -0,0 +1,111 @@
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+
+    List<Cessation> cessations = new ArrayList<>();
+    cessations.addAll(returnIndividualBeneficialOwners(overseasEntitySubmissionDto, combinedBoData));
+    cessations.addAll(returnCorporateEntityBeneficialOwnerCessations(overseasEntitySubmissionDto, combinedBoData));
+    cessations.addAll(returnLegalPersonBeneficialOwners(overseasEntitySubmissionDto, combinedBoData));
+    return cessations;
+  }
+
+  private List<IndividualBeneficialOwnerCessation> returnIndividualBeneficialOwners(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var beneficialOwnersIndividual = overseasEntitySubmissionDto.getBeneficialOwnersIndividual();
+    return beneficialOwnersIndividual.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnIndividualBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private List<CorporateEntityBeneficialOwnerCessation> returnCorporateEntityBeneficialOwnerCessations(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var returned = overseasEntitySubmissionDto.getBeneficialOwnersCorporate();
+    return returned.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnCorporateEntityBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private List<LegalPersonBeneficialOwnerCessation> returnLegalPersonBeneficialOwners(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var returned = overseasEntitySubmissionDto.getBeneficialOwnersGovernmentOrPublicAuthority();
+    return returned.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnLegalPersonBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private IndividualBeneficialOwnerCessation returnIndividualBeneficialOwnerCessation(
+      BeneficialOwnerIndividualDto bo, Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+
+    var hashedId = bo.getChipsReference();
+    var publicPrivateBoPair = Optional.ofNullable(combinedBoData.get(hashedId))
+            .orElseThrow(() -> new IllegalArgumentException(NO_PAIR_FOUND));
```

Comment:
not really an illegal arg exception and what handles the error?

Reply:
we should log an error and skip the cessation for now. Otherwise the fining can never be processed.

Reply:
Ok. If we want to keep the flow going I can make everything Optional for now.

Reply:
Yes, as long as we log an error I think that's preferrable.